DalStatement.php

```diff
@@ -47,6 +47,115 @@
 class DalStatement
 {
     /**
+     * Fetch the next row in the result set.
+     *
+     * @var int
+     */
+    const FETCH_FORWARD      = 0;
+
+    /**
+     * Fetch the previous row in the result set.
+     *
+     * @var int
+     */
+    const FETCH_BACKWARD     = 1;
+
+    /**
+     * Start at the first offset.
+     *
+     * @var int
+     */
+    const FETCH_OFFSET_FIRST = 0;
+
+    /**
+     * Start at the last offset.
+     *
+     * @var int
+     */
+    const FETCH_OFFSET_LAST  = -1;
+
+    /**
+     * Specifies that the fetch method shall return each row as an object with
+     * variable names that correspond to the column names returned in the result
+     * set. FETCH_LAZY creates the object variable names as they are accessed.
+     *
+     * @var int
+     */
+    const FETCH_LAZY  = 1;
```

Comment:
Call to `fetch(FETCH_LAZY)` will return a `\PDORow` object.
It's like with`FETCH_OBJECT` (return `\stdClass`) but there is one difference, the properties value are fetched when the user call it and not at the construct time.